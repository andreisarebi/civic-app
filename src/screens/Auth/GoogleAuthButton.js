import React from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { googleLogin } from '../../auth/redux';
import { getIsLoading } from '../../UI/redux';
import GoogleIcon from './GoogleIcon';
import SocialButton from './SocialButton';
import colors from '../../styles/colors';

const GoogleAuthButton = props => (
  <SocialButton
    textColor={colors.black}
    icon={{}}
    style={props.style}
    buttonStyle={styles.button}
    onPress={props.onPress}
    title={props.title}
    iconComponent={GoogleIcon}
  />
);

GoogleAuthButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  style: SocialButton.propTypes.style,
};

GoogleAuthButton.defaultProps = { style: {} };

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.white,
    borderColor: colors.lightGray2,
  },
});

const mapStateToProps = state => ({ isLoading: getIsLoading(state) });

const mapDispatchToProps = { onPress: googleLogin };

// Prevent monkey clicking the button by disabling the onPress
// method if there's already an in-flight request
const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  onPress: () => !stateProps.isLoading && dispatchProps.onPress(),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps,
)(GoogleAuthButton);
