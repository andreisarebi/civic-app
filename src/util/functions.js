import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
