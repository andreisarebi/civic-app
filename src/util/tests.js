import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
