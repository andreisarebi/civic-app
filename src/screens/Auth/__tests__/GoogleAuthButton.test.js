import { createSnapshot } from '../../../util/tests';
import GoogleAuthButton from '../GoogleAuthButton';

describe('GoogleAuthButton', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(GoogleAuthButton.WrappedComponent, {
      title: 'sign in with google',
      onPress: jest.fn(),
    });
    expect(snapshot).toMatchSnapshot();
  });
});
