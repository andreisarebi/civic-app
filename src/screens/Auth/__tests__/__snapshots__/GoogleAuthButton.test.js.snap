// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleAuthButton renders correctly 1`] = `
<SocialButton
  buttonStyle={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e0e0e0",
    }
  }
  icon={Object {}}
  iconComponent={[Function]}
  onPress={[MockFunction]}
  style={Object {}}
  textColor="#000000"
  title="sign in with google"
/>
`;
